e2e/path/main.go

```diff
@@ -74,6 +74,22 @@ var tests = []test{
 		},
 		expected: "https://fallback-to.path.path.example.com",
 	},
+	{
+		name: "Redirect using the use= field and upstream record",
+		args: data{
+			host: "sourcerecord.path.path.example.com",
+			path: "/testing",
+		},
+		expected: "https://upstream.path.path.example.com",
+	},
+	{
+		name: "Redirect to upstream record using multiple use= fields",
```

Comment:
Might be good to mention testing a non working upstream record and fallthrough to the next use= field. The description doesn't seem to explain the whole test well in my view.